packages/taco-auth/src/providers/eip4361.ts

```diff
@@ -46,7 +64,7 @@ export class EIP4361AuthProvider {
 
   private async createSIWEAuthMessage(): Promise<AuthSignature> {
     const address = await this.signer.getAddress();
-    const { domain, uri } = this.getParametersOrDefault();
+    const { domain, uri } = this.providerParams;
     const version = '1';
     const nonce = generateNonce();
     const chainId = (await this.provider.getNetwork()).chainId;
```

Comment:
Can we do `signer.getChainId( )` here and get rid of `provider` from the constructor?

Reply:
Not every `signer` has a connected `provider`. In `ethers`, it's possible to create a signer using a raw private key without ever connecting to an RPC. 

I think we may want to explore requiring that every `signer` has a `provider` already connected and to do that in the whole codebase. But I think this is for another PR since we would be changing quite a lot of APIs.

Reply:
Ah ok.